core/include/scipp/core/except.h

```diff
@@ -169,8 +169,47 @@ template <class T> void countsOrCountsDensity(const T &object) {
 }
 
 void SCIPP_CORE_EXPORT validSlice(const Dimensions &dims, const Slice &slice);
-void SCIPP_CORE_EXPORT coordsAndLabelsMatch(const DataConstProxy &a,
-                                            const DataConstProxy &b);
+
+template <typename A, typename B>
+void coordsAndLabelsMatch(const A &a, const B &b) {
```

Comment:
Previously this function work for `DataConstProxy`, now it would not compile anymore? Probably not a problem right now...? What is used for checking coords in operations with `DataProxy`?